src/texture.rs

```diff
@@ -0,0 +1,109 @@
+use std::cell::RefCell;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::BTreeMap;
+use std::hash::Hasher;
+use std::ops::Deref;
+use std::rc::Rc;
+
+use super::{ImStr, ImTextureID};
+
+/// Trait that an object representing a texture usable by the drawing back-end
+/// should implement.
+pub trait ImTexture {
+    /// Get [`ImTextureID`]
+    fn get_id(&self) -> ImTextureID;
+    /// Query texture size, in pixels (for convenience)
+    fn get_size(&self) -> (u32, u32);
+}
+
+/// We implement [`ImTexture`] for [`ImTextureID`] one only to be able to use
+/// it on the ID of the font texture (as is).
+impl ImTexture for ImTextureID {
+    fn get_id(&self) -> ImTextureID { *self }
+    fn get_size(&self) -> (u32, u32) {
+        panic!("Cannot get size of ImTextureID (it is a raw pointer)!")
```

Comment:
A convenience function that panics sounds like a bad idea. Do we really need `get_size` in the trait if we can't implement it for the font texture?